.drone.star

```diff
@@ -527,8 +547,35 @@ def coreApiTests(ctx, part_number = 1, number_of_parts = 1, storage = "owncloud"
         "volumes": [pipelineVolumeOC10Tests],
     }
 
+def apiTests(ctx, pipelines):
+    partsToSkip = config["apiTests"]["skipExceptParts"]
+    for runPart in range(1, config["apiTests"]["numberOfParts"] + 1):
+        if len(partsToSkip) != 0:
+            if runPart in partsToSkip:
+                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "owncloud"))
+                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "ocis"))
+        else:
+            pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "owncloud"))
+            pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "ocis"))
```

Comment:
Aren't these exactly the same?

Reply:
Nope,
it conditionally appends pipelines.
If I want to run only part 5 of 10 (numberOfParts) with `skipExceptParts: [5]` then `apiTests()` will append only that pipeline.
Else the whole pipelines (1-10 parts) are appended.